frontend/src/v5/ui/controls/assignees/assigneesSelect/assigneeSelect.component.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { selectJobs } from '@/v4/modules/jobs';
+import { UsersHooksSelectors } from '@/v5/services/selectorsHooks/usersSelectors.hooks';
+import { Assignees } from '@controls/assignees/assignees.component';
+import { MultiSelectMenuItem } from '@controls/formMultiSelect/multiSelectMenuItem/multiSelectMenuItem.component';
+import { SearchSelect } from '@controls/searchSelect/searchSelect.component';
+import { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { HiddenSearchSelect } from './assigneeSelect.styles';
+
+type AssigneesSelectProps = {
+	values: string[];
+	onBlur: (values) => void;
+};
+
+export const AssigneesSelect = ({ values: initialValues, onBlur, ...props }: AssigneesSelectProps) => {
+	const [values, setValues] = useState(initialValues);
+	const [open, setOpen] = useState(false);
+
+	const users = UsersHooksSelectors.selectCurrentTeamspaceUsers();
+	const jobs = useSelector(selectJobs);
+	const allUsersAndJobs = users.map(({ user, firstName, lastName }) => ({ value: user, label: `${firstName} ${lastName}` }));
+	Array.prototype.push.apply(allUsersAndJobs, jobs.map(({ _id }) => ({ value: _id, label: _id })));
+
+	const preventPropagation = (e) => {
+		if (e.key !== 'Escape') e.stopPropagation();
+	};
+	const onClick = (e) => {
+		preventPropagation(e);
+		setOpen(true);
+	};
+	const handleClose = (e) => {
+		preventPropagation(e);
+		setOpen(false);
+		onBlur(values);
+	};
+	const onChange = (e) => {
+		preventPropagation(e);
```

Comment:
why is the preventPropagation bit necessary in all instances?

Reply:
It's necessary in onClick and handleClose because otherwise opening and closing the assignee select can open the ticket details. It's not actually needed any more in the onChange nor in the dueDateInput onClickBackdrop, so I will remove them